app/Console/Command/MonitorWorkersShell.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+declare(strict_types=1);
+
+App::uses('BackgroundJobsTool', 'Tools');
+
+class MonitorWorkersShell extends AppShell
+{
+    /** @var BackgroundJobsTool */
+    private $BackgroundJobsTool;
+
+    /** @var int */
+    private $sleepInterval;
+
+    private const DEFAULT_SLEEP_INTERVAL = 5; // seconds
+
+    public $tasks = ['ConfigLoad'];
+
+    public function initialize(): void
+    {
+        parent::initialize();
+        $this->BackgroundJobsTool = new BackgroundJobsTool(Configure::read('SimpleBackgroundJobs'));
+    }
+
+    public function getOptionParser(): ConsoleOptionParser
+    {
+        $parser = parent::getOptionParser();
+        $parser->addOption(
+            'sleep',
+            [
+                'help' => 'Sleep interval between jobs (seconds).',
+                'default' => self::DEFAULT_SLEEP_INTERVAL,
+                'required' => false
+            ]
+        );
+
+        return $parser;
+    }
+
+    public function main(): void
+    {
+        $this->sleepInterval = (int)$this->params['sleep'];
+        CakeLog::info("[WORKERS MONITOR] - starting to monitor workers...");
+
+        while (true) {
+            $this->ConfigLoad->execute();
+            $this->checkWorkersProcessStatus($this->BackgroundJobsTool->getWorkers());
+
+            sleep($this->sleepInterval);
+        }
+    }
+
+    /**
+     * Check workers process status
+     *
+     * @param Worker[] $workers
+     * @return void
+     */
+    private function checkWorkersProcessStatus(array $workers): void
+    {
+        foreach ($workers as $worker) {
+            if (!file_exists("/proc/{$worker->pid()}")) {
```

Comment:
I think this will not work, because the result of this function is cached.

Reply:
not sure I understand, could you elaborate a bit more?
By cached you mean its only updated every `sleepInterval` seconds?

Reply:
Check https://www.php.net/clearstatcache

Reply:
true, i'll refactor to rely only on Supervisor API for workers info/mgmt